libcaf_net/caf/net/ip.hpp

```diff
@@ -0,0 +1,36 @@
+/******************************************************************************
+ *                       ____    _    _____                                   *
+ *                      / ___|  / \  |  ___|    C++                           *
+ *                     | |     / _ \ | |_       Actor                         *
+ *                     | |___ / ___ \|  _|      Framework                     *
+ *                      \____/_/   \_|_|                                      *
+ *                                                                            *
+ * Copyright 2011-2018 Dominik Charousset                                     *
+ *                                                                            *
+ * Distributed under the terms and conditions of the BSD 3-Clause License or  *
+ * (at your option) under the terms and conditions of the Boost Software      *
+ * License 1.0. See accompanying files LICENSE and LICENSE_ALTERNATIVE.       *
+ *                                                                            *
+ * If you did not receive a copy of the license files, see                    *
+ * http://opensource.org/licenses/BSD-3-Clause and                            *
+ * http://www.boost.org/LICENSE_1_0.txt.                                      *
+ ******************************************************************************/
+
+#pragma once
+
+#include <cstddef>
+#include <string>
+
+namespace caf {
+namespace net {
+
+// IP version tag.
+enum class ip { v4, v6 };
```

Comment:
We only need this for the `preferred` arguments. I hope we can get rid of these to also make this header obsolete.

Reply:
The argument currently has three states. Prefer IPv4, prefer IPv6, and don't prefer anything. Since it already is optional we could just as well make it an optional bool named `prefer_v6` that is set to `none`. Or are you suggesting to change the functionality here?

Reply:
Probably solves itself if we find a better way to address the comment above.